visualmetrics.py

```diff
@@ -793,16 +841,18 @@ def main():
                 orange_file = None
                 if options.orange:
                     orange_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'orange.png')
-                if options.orange and\
-                    not os.path.isfile(orange_file):
+                if options.orange and \
+                        not os.path.isfile(orange_file):
                     orange_file = os.path.join(temp_dir, 'orange.png')
                     generate_orange_png(orange_file)
                 video_to_frames(options.video, directory, options.force, orange_file, options.viewport, options.full,
                                 options.timeline)
             calculate_histograms(directory, histogram_file, options.force)
-            if options.dir is not None and options.quality is not None:
+            # Perceptual SI computation works on png's only
+            if options.dir is not None and options.quality is not None and options.perceptual is False:
```

Comment:
What about the case where video is provided so png's are created for intermediate processing but jpeg is the end state?  The Perceptual SI should be able to operate on the intermediate png's but still allow final step conversion to jpeg.

Maybe swap the order and calculate the visual metrics before jpeg conversion.